### PR TITLE
feat(react): add useErrorBoundary

### LINKED ISSUE #13: [Feature]: add useErrorBoundary
### **Package Scope**
react

- [x] Add to an existing package
- [ ] New package

Package name: react

### PR DESCRIPTION
**Pull request by manudeli**
fix #13 

# Overview

I added `useErrorBoundary`. this Pull Request is to add experimental minor feature
so I hope you @okinawaa care conservatively new interface name like `reset`, `setError`, etc. of useErrorBoundary

## what I think for requirement of useErrorBoundary 
1. useErrorBoundary need to `assert ErrorBoundary in parent`
2. useErrorBoundary need to be used in `ErrorBoundary.children and ErrorBoundary.fallback both`
3. boolean value all have to be named with prefix `is` like `isError` easily

## PR Checklist

- [x] I have written documents and tests, if needed.

### CODE REVIEW

**Comment by manudeli** [comment]
@minsoo-web could you review this please as one of user? this is new feature of @suspensive/react, I need careful review all code before bump minor version

File: packages/react/src/ErrorBoundary.spec.tsx [hunk 1]
```diff
@@ -225,3 +226,99 @@ describe('withErrorBoundary', () => {
...
+    expect(container.textContent).toBe(ERROR_MESSAGE)
+    expect(container.textContent).not.toBe(TEXT)
+    expect(onReset).toHaveBeenCalledTimes(0)
+    act(() => vi.advanceTimersByTime(MS_100))
+    expect(container.textContent).toBe(TEXT)
+    expect(container.textContent).not.toBe(ERROR_MESSAGE)
+    expect(onReset).toHaveBeenCalledTimes(1)
+  })
+
+  it('should supply setError to set Error of ErrorBoundary manually', () => {
```
Comment (okinawaa) [inline packages/react/src/ErrorBoundary.spec.tsx:279]: 👍

File: packages/react/src/ErrorBoundary.spec.tsx [hunk 2]
```diff
@@ -225,3 +226,99 @@ describe('withErrorBoundary', () => {
...
+    ThrowError.reset()
+  })
+  const renderErrorBoundary = (props: Partial<ComponentProps<typeof ErrorBoundary>>) =>
+    act(() =>
+      root.render(
+        <ErrorBoundary ref={errorBoundaryRef} fallback={(caught) => <>{caught.error.message}</>} {...props} />
+      )
+    )
+
+  it('should supply reset function to reset in fallback of ErrorBoundary', () => {
```
Comment (okinawaa) [inline packages/react/src/ErrorBoundary.spec.tsx:243]: 👍

File: packages/react/src/ErrorBoundary.en.mdx [hunk 3]
```diff
@@ -58,3 +58,44 @@ const Example = () => {
...
+
+  return <>...</>
+}
+
+const Example = () => (
+  <ErrorBoundary fallback={ErrorBoundaryFallback}>
+    <SetErrorAfterFetch />
+  </ErrorBoundary>
+)
+```
```
Comment (okinawaa) [inline packages/react/src/ErrorBoundary.en.mdx:101]: when I see this doc 
clearly understood thx 👍

I think interface is good

File: packages/react/src/ErrorBoundary.en.mdx [hunk 4]
```diff
@@ -58,3 +58,44 @@ const Example = () => {
...
+    <Error />
+  </ErrorBoundary>
+)
+```
+
+We can use useErrorBoundary.setError in ErrorBoundaryFallback to make ErrorBoundary aware of the Error without throw.
+
+```tsx
+const SetErrorAfterFetch = () => {
+  const errorBoundary = useErrorBoundary()
```
Comment (okinawaa) [inline packages/react/src/ErrorBoundary.en.mdx:84]: In fact, when using the `useErrorboundary`, I used it put errors that occur other than rendering into the errorboundary. I think it is good to added in the document

[reference](https://slash.page/libraries/react/error-boundary/src/errorboundary.i18n/#useerrorboundary)
>React's [error boundary](https://reactjs.org/docs/error-boundaries.html) component cannot catch an error occurred in event handlers, asynchronous code like setTimeout, etc.
useErrorBoundary is useful to deliver an error to the nearest error boundary in anywhere.

**Review by okinawaa — APPROVED** [review]

File: packages/react/src/ErrorBoundary.en.mdx (hunk 4 above)
Comment (manudeli) [reply packages/react/src/ErrorBoundary.en.mdx:84]: Cool👍, Could you add it more please after merging this Pull Request

**Comment by manudeli** [comment]
@okinawaa glad to release new feature. I marked it as experimental. Let's make it better gradually

**Pull request merged**